Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -184,15 +184,15 @@ AppDelegateViewModelOutputs {
 
   // swiftlint:disable cyclomatic_complexity
   public init() {
-    let currentUserEvent = Signal
+    let currentUserEvent: Signal<Signal<User?, ErrorEnvelope>.Event, NoError> = Signal
```

Comment:
Might not need the whole type signature here but it's cool if it helps the compiler along, it should be able to infer it from the type in the `switchMap` I think.

Reply:
Yeah, you are right. Since the compiler was complaining so much and not recognising types, I decided to be more conservative and explicitly define the type signature instead of relying on the type inference. But it worked fine after I removed the signature of line 187.